### PR TITLE
feat: add support for variable where criteria expressions

### PR DESCRIPTION
**Pull request by igdianov**
This PR adds support for variable where criteria expressions, i.e. given

```
query($where: BooksCriteriaExpression) {
  Books (where: $where) {
    select {
      id(orderBy: ASC)
      title
      genre
    }
  }
}
```
with variables payload:

```
{
  "where": { 
    "genre": {
      "IN": ["NOVEL"]
    }
  }
}
```
the expected result will be:

```
{
  "data": {
    "Books": {
      "select": [
        {
          "id": 2,
          "title": "War and Peace",
          "genre": "NOVEL"
        },
        {
          "id": 3,
          "title": "Anna Karenina",
          "genre": "NOVEL"
        }
      ]
    }
  }
}
```
Fixes https://github.com/introproventures/graphql-jpa-query/issues/126

### CODE REVIEW

**Comment by igdianov** [comment]
@anotender Here is the first crack of the support passing where criteria expressions via variables. Feel free to try adding any test cases for your requirements...

File: graphql-jpa-query-schema/src/main/java/com/introproventures/graphql/jpa/query/schema/impl/QraphQLJpaBaseDataFetcher.java [hunk 1]
```diff
@@ -897,7 +918,29 @@ else if (value instanceof VariableReference) {
             }
         } else if (value instanceof ArrayValue) {
             Object convertedValue =  environment.getArgument(argument.getName());
-            if (convertedValue != null && !getJavaType(environment, argument).isEnum()) {
+
```
Comment (anotender) [inline graphql-jpa-query-schema/src/main/java/com/introproventures/graphql/jpa/query/schema/impl/QraphQLJpaBaseDataFetcher.java]: @igdianov what was the purpose of this change. I tried to run all the test with and without this change and they passed. Maybe we should add a test case for the issue solved by this change?

Comment (igdianov) [reply graphql-jpa-query-schema/src/main/java/com/introproventures/graphql/jpa/query/schema/impl/QraphQLJpaBaseDataFetcher.java]: @anotender The test case for this change already exists. I made the change to address the conversion of enum values arrays passed in the variable payload. i.e.

```
"genre": {
   "IN": ["NOVEL"]
 }
```

For some reason, Graphql Ast converter does not do that. I don't like the code, so I will try to refactor it to make it better.

**Comment by anotender** [comment]
@igdianov as for me, the changes look ok at the moment. Do you want to add something more?
And one more thing, what formatter should I use? As the default Intellij one seems to not format files as it should.

**Comment by igdianov** [comment]
@anotender Thanks! Everything looks good. I will merge and release this on the weekend. I am using STS 4.0 with custom formatter. I don't know how to export to IDEA. I will try to find a matching IDEA configuration.

**Pull request merged**